### ISSUE TITLE
Roads sometimes don't connect properly

### ISSUE THREAD

**Issue by Forecaster** [citybound/citybound#132]
This was a simple loop and according to the plan it was connected fine. But you can clearly see it's split into two separate segments that are not connected.

http://imgur.com/a/3gUrw

**Comment by Forecaster** [comment]
Selecting works across the right point though, even though the disconnected symbols appear there there seems to be a proper connection.

**Comment by Forecaster** [comment]
Also something else interesting happened with this road...

https://youtu.be/7NB4Zi6ubag

**Comment by aeickhoff** [comment]
Yeah, loops are a pretty special case in the code. The disconnected symbols on the right mean that merging between the lanes won't work. Thanks again for documenting all of that so nicely.
